starlette/routing.py

```diff
@@ -239,7 +239,9 @@ def matches(self, scope: Scope) -> typing.Tuple[Match, Scope]:
                     return Match.FULL, child_scope
         return Match.NONE, {}
 
-    def url_path_for(self, name: str, **path_params: typing.Any) -> URLPath:
+    def url_path_for(self, *args: str, **path_params: typing.Any) -> URLPath:
+        assert len(args) == 1, "url_path_for() takes exactly one positional argument"
```

Comment:
Let's raise `TypeError` on incorrect usages, rather than `assert`.

Reply:
Fixed in 1892edd.